### PR TITLE
30 search for rooms

### PR DESCRIPTION
**Pull request by T-z**

### CODE REVIEW

**Comment by asiday** [comment]
CR_1

**Comment by asiday** [comment]
Here is my test story

I have two accommodations. The room20m2 is available from 1st November and room50m2 is available from 1st December.
 

When I select avalability from 1stNovember ->


It shows corecctly


When I select avalability from 1st December ->


The answer is also correct


But if I select from the middle of November, I expect that room50m2 will be shown


But there is no rooms

File: am-ui/webcomponents/search-webcomponent/src/assets/i18n/de.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ export default {
     offers: 'Angebote',
     searches: 'Anfragen',
     select: 'Art der Unterkunft',
+    furnished: 'Möbiliert',
     categories: {
```
Comment (asiday) [inline am-ui/webcomponents/search-webcomponent/src/assets/i18n/de.ts:8]: Typo: "Möbliert"

File: am-ui/webcomponents/search-webcomponent/src/assets/i18n/de.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ export default {
     offers: 'Angebote',
     searches: 'Anfragen',
     select: 'Art der Unterkunft',
+    furnished: 'Möbiliert',
```
Comment (asiday) [inline am-ui/webcomponents/search-webcomponent/src/assets/i18n/de.ts:7]: Typo: "Möbliert"

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java [hunk 3]
```diff
@@ -337,6 +369,45 @@ private List<TopicSearchResponseDTO> filterTopicsByToPrice(String topicValueToPr
         return topicList;
     }
 
+    private List<TopicSearchResponseDTO> filterTopicsByFromDate(String topicValueFromDate, 
+                   List<TopicSearchResponseDTO> topicList) {
+        
+        String[] fromDate = topicValueFromDate.split("-", 3);
```
Comment (asiday) [inline marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java:375]: May be it is better to use here the date API - LocalDate. 
Something like this:
LocalDate date = LocalDate.parse(topicValueFromDate");

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java [hunk 4]
```diff
@@ -337,6 +369,45 @@ private List<TopicSearchResponseDTO> filterTopicsByToPrice(String topicValueToPr
...
+        try {
+
+        return topicList
+        .stream()
+        .filter(topic -> topic.getTopicValues()
+        .stream()
+        .filter(topicValue -> topicValue.getFieldId() == DATE)
+        .anyMatch(topicValue -> 
+          Long.parseLong(topicValue.getValue().split("-", 3)[0]) >= Long.parseLong(fromDate[0]) && Long.parseLong(topicValue.getValue().split("-", 3)[1]) >= Long.parseLong(fromDate[1]) && Long.parseLong(topicValue.getValue().split("-", 3)[2]) >= Long.parseLong(fromDate[2])
```
Comment (asiday) [inline marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java:384]: Looks complicated. Please use LocalDate when working with dates.

File: am-ui/webcomponents/search-webcomponent/src/app/search-form/search-form.component.scss [hunk 5]
```diff
@@ -113,4 +113,18 @@
   .field-container {
     overflow: auto;
   }
+
+  .furnished {
+    display: flex;
+    align-items: center;
+
+    >input{
+      width: 1em;
```
[the remark this answers is not in the document]
Comment (T-z) [reply am-ui/webcomponents/search-webcomponent/src/app/search-form/search-form.component.scss:122]: It is just an SCSS statement. It means direct "input" tag after the parent bloc.

**Review by asiday — APPROVED** [review]

File: marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java (hunk 3 above)
Comment (T-z) [reply marketplace-service/src/main/java/ch/akros/marketplace/service/service/TopicService.java]: You're right. the code was rewritten. Thanks for the advice.

**Pull request merged**